lib/iris/fileformats/netcdf/saver.py

```diff
@@ -2454,64 +2345,175 @@ def _lazy_stream_data(data, fill_value, fill_warn, cf_var):
+        result can be computed, otherwise computation will hang (never return).
+        """
+        if self._delayed_writes:
+            # Create a single delayed da.store operation to complete the file.
+            sources, targets, fill_infos = zip(*self._delayed_writes)
+            store_op = da.store(sources, targets, compute=False, lock=False)
+
+            # Construct a delayed fill-check operation for each (lazy) source array.
+            delayed_fillvalue_checks = [
+                # NB with arraylib=dask.array, this routine does lazy array computation
+                _data_fillvalue_check(da, source, fillinfo.check_value)
+                for source, fillinfo in zip(sources, fill_infos)
+            ]
+
+            # Return a single delayed object which completes the delayed saves and
+            # returns a list of any fill-value warnings.
+            @dask.delayed
+            def compute_and_return_warnings(store_op, fv_infos, fv_checks):
+                # Note: we don't actually *do* anything with the 'store_op' argument,
+                # but including it here ensures that dask will compute it (thus
+                # performing all the delayed saves), before calling this function.
+                results = []
+                # Pair each fill_check result (is_masked, contains_value) with its
+                # fillinfo and construct a suitable Warning if needed.
+                for fillinfo, (is_masked, contains_value) in zip(
+                    fv_infos, fv_checks
+                ):
+                    fv_warning = _fillvalue_report(
+                        fill_info=fillinfo,
+                        is_masked=is_masked,
+                        contains_fill_value=contains_value,
+                    )
+                    if fv_warning is not None:
+                        # Collect the warnings and return them.
+                        results.append(fv_warning)
+                return results
+
+            result = compute_and_return_warnings(
+                store_op,
+                fv_infos=fill_infos,
+                fv_checks=delayed_fillvalue_checks,
+            )
+
+        else:
+            # Return a delayed, which returns an empty list, for usage consistency.
+            @dask.delayed
+            def no_op():
+                return []
+
+            result = no_op()
+
+        return result
+
+    def complete(self, issue_warnings=True) -> List[Warning]:
+        """
+        Complete file by computing any delayed variable saves.
+
+        This requires that the Saver has closed the dataset (exited its context).
+
+        Parameters
+        ----------
+        issue_warnings : bool, default = True
+            If true, issue all the resulting warnings with :func:`warnings.warn`.
+
+        Returns
+        -------
+        warnings : list of Warning
+            Any warnings that were raised while writing delayed data.
+
+        """
+        if self._dataset._isopen:
```

Comment:
```suggestion
        if self._dataset.isopen():
```
Just spotted : this is not good usage, use the public API form instead.